backend/src/layer/data/flickr_api/json_parser.rs

```diff
@@ -54,18 +54,26 @@ impl JsonParser {
     /// # Return
     /// A boolean if the image has an valid license or not.
     /// If the image has no license or no license history, the image isn't restricted by any license and true will be returned.
-    #[must_use]
-    pub fn check_license(root: &JsonRootLicense) -> (bool, String) {
+    /// # Errors
+    /// If an image is invalid [`ImageHosterError::InvalidLicense`] with the invalid license will be returned.
+    pub fn check_license(root: JsonRootLicense) -> Result<(), ImageHosterError> {
         let license = root
             .license_history
-            .iter()
+            .into_iter()
             .max_by_key(|l| l.date_change)
-            .map(|entry| &entry.new_license);
+            .map(|entry| entry.new_license);
 
         if let Some(license) = license {
-            return (VALID_LICENSES.contains(&license.as_str()), license.clone());
+            let str_license = license.as_str();
+            return if VALID_LICENSES.contains(&str_license) {
+                Ok(())
+            } else {
+                Err(ImageHosterError::InvalidLicense(license))
+            };
         }
-        (false, String::from("No license could be detected"))
+        Err(ImageHosterError::InvalidLicense(String::from(
+            "No license could be detected",
+        )))
```

Comment:
reorder: put option with `.ok_or(Error...)?` first and return it.
then no `return` is needed as it all can just be the last line.